src/shogun/classifier/Perceptron.cpp

```diff
@@ -39,10 +39,7 @@ void CPerceptron::init_model(CFeatures* data)
 		if (!data->has_property(FP_DOT))
 			SG_ERROR("Specified features are not of type CDotFeatures\n")
 		set_features((CDotFeatures*) data);
-
-		SG_REF(data);
-		SG_UNREF(m_continue_features);
-		m_continue_features = data->as<CDotFeatures>();
+		set_continue_features(data);
```

Comment:
could we make this happen in the IterativeMachine class without the dev having to specify this in every use of the mixin?